app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1796,9 +1843,25 @@ public void showView(View next, AnimationType from, boolean animateLastView) {
         mViewPane.addView(mCurrentView, lp);
 
         mCurrentView.startAnimation(mInAnimation);
-        if (mCurrentView instanceof ODKView)
+        
+        if (mCurrentView instanceof ODKView) {
             ((ODKView) mCurrentView).setFocus(this);
-        else {
+            
+            String groupLabelText = ((ODKView) mCurrentView).getGroupLabel();
+
+            FrameLayout header = (FrameLayout)mCurrentView.findViewById(R.id.form_entry_header);
+
+            TextView groupLabel = ((TextView)header.findViewById(R.id.form_entry_group_label));
+
+            header.setVisibility(View.GONE);
+            groupLabel.setVisibility(View.GONE);
+
+            if(groupLabelText != "") {
```

Comment:
Never use ==/!= with Strings (also always use a linter, it will warn you of this). 
This shouldn't work unless you change it to !"".equals(groupLabelText)